modin/experimental/xgboost/xgboost.py

```diff
@@ -76,6 +117,200 @@ def __iter__(self):
         yield self.data
         yield self.label
 
+    def get_dmatrix_params(self):
+        """
+        Forming a dictionary from parameters other than label and date.
+
+        Returns
+        -------
+        Dictionary from kwargs of dmatrix.
+        """
+        dmatrix_params = {
+            "feature_names": self.feature_names,
+            "feature_types": self.feature_types,
+            "missing": self.missing,
+            "silent": self.silent,
+            "feature_weights": self.feature_weights,
+            "enable_categorical": self.enable_categorical,
+        }
+        return dmatrix_params
+
+    @property
+    def feature_names(self):
+        """
+        Get feature numes (column labels).
+
+        Returns
+        -------
+        Value parameter ``feature_names`` of dmatrix.
+        """
+        return self._feature_names
+
+    @feature_names.setter
+    def feature_names(self, feature_names):
+        """
+        Set feature names (column labels).
+
+        Parameters
+        ----------
+        feature_names : list or None
+            Labels for features. None will reset existing feature names
+
+        Returns
+        -------
+        Add in dmatrix new value of parameter ``feature_types``.
+        """
+        if feature_names is not None:
+            # validate feature name
+            try:
+                if not isinstance(feature_names, str):
+                    feature_names = list(feature_names)
+                else:
+                    feature_names = [feature_names]
+            except TypeError:
+                feature_names = [feature_names]
```

Comment:
What's is the need of this block? Give an explanation please

Reply:
This was changed